packages/core/src/networks.ts

```diff
@@ -140,6 +142,12 @@ export const calculateMerkleLeafGas = (chainId: bigint, foundryGas: string) => {
   const network = SPHINX_NETWORKS.find((n) => n.chainId === chainId)
 
   if (network?.hardcodedMerkleLeafGas) {
+    if (BigInt(foundryGas) > BigInt(network?.hardcodedMerkleLeafGas)) {
+      throw new InvariantError(
+        'Transaction too large to be executed on Pangolin'
```

Comment:
I don't think this should be an `InvariantError` because this type of error is appended with "Should never happen". It seems like this error will be triggered if the user has a very large transaction